### DISCUSSION TITLE
Atomic updates and power loss safety

### DISCUSSION THREAD

**Discussion by jmpolom** [rancher/elemental-toolkit#2091] [General]
Are elemental updates fully atomic (IE: they either succeed completely or they do not) with a guarantee that an update will never only partially complete? Are updates safe against power loss during an update process likely by relying on the same update atomicity?

**Comment by frelon** [comment]
Hi @jmpolom!

Yes, updates are atomic and are written to a new snapshot and then rebooted into the upgraded system. Any powerloss during upgrade should boot into the working system and a new upgrade should just work!

Also with the new btrfs snapshotter any corruption should also be detected and repaired automatically.
